default/AI/freeorion_tools.py

```diff
@@ -26,9 +27,9 @@ def get_ai_tag_grade(tag_list, tag_type):
     X is most commonly (but not necessarily) one of [NO, BAD, AVERAGE, GOOD, GREAT, ULTIMATE]
     If no matching tags, returns empty string (which for most types should be considered equivalent to AVERAGE)
     '''
-    for tag in tag_list:
+    for tag in filter(lambda tag: tag.startswith("AI_TAG_"), tag_list):
         parts = tag.split("_")
-        if parts[:2] == ["AI","TAG"] and [x.upper() for x in parts[3:4]] == [tag_type.upper()]:
+        if parts[3:4] == [tag_type.upper()]:
```

Comment:
I agree your second point is a good suggestion.
Regarding your first point, I'm quite sure you are right that all the current calls have tag_type already being uppercase.  But once again, keep in mind the home-modder-- FOCS is intended to be case-independent, and while we can't avoid some case-dependency in python, I don't want to add any where it is not necessary.